File: SieuSoNguyenTo.py
import math

def solution(b):
    isPrime = [True] * (b + 1)
    if b < 0:
        return False
    
    isPrime[0] = False
    if b + 1 > 1:
        isPrime[1] = False
    count = 0
    for i in range(2, round(math.sqrt(b)) + 1):
        if isPrime[i]:
            for j in range(i * i, b + 1, i):
                isPrime[j] = False

    for i in range(2, b):
        if isPrime[i]:
            count += 1
            
    return isPrime[count]

def main():
    a = int(input())
    b = int(input())
    count = 0
    for i in range(a, b + 1):
        if solution(i):
            count += 1
    print(count)            
                                    

if __name__ == '__main__':
    main()
    

File: SieuSoNguyenTo.java
package Security.coPhuongWriteUp;

import java.util.Scanner;

public class SieuSoNguyenTo {

    static Scanner scanner = new Scanner(System.in);

    public static boolean solution(int n) {
        Boolean[] isPrime = new Boolean[n + 1];
        if (n < 0) {
            return false;
        }
        isPrime[0] = false;
        if (n + 1 > 1) {
            isPrime[1] = false;
        }
        for (int i = 2; i < isPrime.length; i++) {
            isPrime[i] = true;
        }

        for (int i = 2; i*i <= n; i++) {
            if (isPrime[i]) {
                for (int j = i*i; j <= n; j+= i) {
                    isPrime[j] = false;
                }
            }
        }

        int count = 0;

        for (int i = 2; i < n; i++) {
            if (isPrime[i]) {
                count++;
            }
        }

        return isPrime[count];
    }


    public static void main(String[] args) {
        int a = scanner.nextInt();
        int b = scanner.nextInt();

        int count = 0;
        for (int i = a; i <= b; i++) {
            if (solution(i)) {
                count++;
            }
        }

        System.out.println(count);
    }
    
}
